samples/Basic/Basic.WebSvc/Controllers/QueryController.cs

```diff
@@ -37,7 +38,149 @@ public Task<IHttpActionResult> GetMetadata(string application, string service)
+			[FromBody] ValueViewModel[] obj)
+		{
+			return base.DeleteAsync(application, service, collection, obj);
+		}
+
+		/// <summary>
+		/// Adds appropriate key & corresponding value to the given reliable collection in the queryable service.
+		/// SINGLE ADD:
+		/// - POST /query/BasicApp/ProductSvc/products and in Body provide a Json:
+		/// In Body: [{
+		///	        "Key": "sku-218",
+		///	        "Value":  {
+		///	            "Sku": "sku-218",
+		///	            "Price": 10.95,
+		///	            "Quantity":46
+		///	        },
+		///	        "PartitionId": "946fd004-37aa-4ea6-94a0-3013d8956fef"
+		///	    }
+		///	    ]
+		/// Record belonging to the key provided in the JSON Body of HTTP POST Request is added to a partition ID mentioned, if its not existing already.
+		/// Incase Partition ID is not mentioned, Record is added to random partition ID.
+		/// 
+		/// 
+		/// BATCH ADD (Provide an Array of keys & values with optional partitionID in JSON format inside body of HTTP POST request to add them all.). 
+		/// -POST /query/BasicApp/ProductSvc/products
+		/// In Body: [{
+		///             "Key": "sku-218",
+		///             "PartitionId": "946fd004-37aa-4ea6-94a0-3013d8956fef"
+		///           },{
+		///             "Key": "sku-217",
+		///             "PartitionId": "946fd004-37aa-4ea6-94a0-3013d8956fef"
+		///          }]
+		/// 
+		/// ADD a Key & Value to a random Partition:
+		/// -POST /query/BasicApp/ProductSvc/products 
+		/// In Body : 
+		/// [{
+		///"Key": "sku-218"
+		///}]
+		/// 
+		/// </summary>
+		[HttpPost]
+		[Route("query/{application}/{service}/{collection}")]
+		public Task<IHttpActionResult> Add(string application, string service, string collection,
+			[FromBody] ValueViewModel[] obj)
+		{
+			return base.AddAsync(application, service, collection, obj);
+		}
+
+		/// <summary>
+		/// Updates appropriate key & corresponding value to the given reliable collection in the queryable service.
+		/// SINGLE UPDATE:
+		/// - PUT /query/BasicApp/ProductSvc/products and in Body provide a Json:
+		/// In Body: [{
+		///	        "Key": "sku-217",
+		///	        "Value":  {
+		///	            "Sku": "sku-217",
+		///	            "Price": 11.95,
+		///	            "Quantity":40
+		///	        },
+		///	        "PartitionId": "946fd004-37aa-4ea6-94a0-3013d8956fef"
+		///	    }
+		///	    ]
+		/// Record belonging to the key provided in the JSON Body of HTTP PUT Request is loacted in the partition ID mentioned & is updated with the new value, if its not existing already then its added..
+		/// Incase Partition ID is not mentioned, Records with matching key from all partitions are located & updated with the new value.
+		/// 
+		/// 
+		/// BATCH UPDATE (Provide an Array of keys & values with optional partitionID in JSON format inside body of HTTP PUT request to update all the keys with new values.). 
+		/// -PUT /query/BasicApp/ProductSvc/products
```

Comment:
I'd like to see a URI for each entity, rather than having it partially encoded in the URL and partially in the body. `PUT /query/BasicApp/ProductSvc/products/39704134134861/sku-218` feels more user-friendly.